src/Pelagos/Entity/DatasetSubmission.php

```diff
@@ -1689,6 +1707,50 @@ public function getDatasetFileSha256Hash()
         return $this->datasetFileSha256Hash;
     }
 
+    /**
+     * Set the hash of the single archive file to be stored in cold storage.
+     *
+     * @param string $datasetFileColdStorageArchiveSha256Hash Hash of the archive to be put into cold storage.
+     *
+     * @return void
+     */
+    public function setDatasetFileColdStorageArchiveSha256Hash($datasetFileColdStorageArchiveSha256Hash)
+    {
+        $this->datasetFileColdStorageArchiveSha256Hash = $datasetFileColdStorageArchiveSha256Hash;
+    }
+
+    /**
+     * Get the hash of the single archive file to be stored in cold storage.
+     *
+     * @return string
+     */
+    public function getDatasetFileColdStorageArchiveSha256Hash()
+    {
+        return $this->datasetFileColdStorageArchiveSha256Hash;
+    }
+
+    /**
+     * Set the size of the single archive file to be stored in cold storage.
+     *
+     * @param integer $datasetFileColdStorageArchiveSize The archive size, in bytes.
+     *
+     * @return void
+     */
+    public function setDatasetFileColdStorageArchiveSize($datasetFileColdStorageArchiveSize)
```

Comment:
Should it be initialized to null? because not all datasetSubmission objects would have this value and it would throw an error (must be an instance of integer and null given)

Reply:
If copying from another dataset submission, it may or may not be null. I changed the docblock to allow for nulls.